src/java/org/apache/cassandra/service/StorageServiceMBean.java

```diff
@@ -46,6 +46,7 @@
      */
     public enum NonTransientError {
         COMMIT_LOG_CORRUPTION,
+        EXCEEDED_DISK_THRESHOLD,
```

Comment:
So this might be slightly annoying, but technically, this _isn't_ a non-transient error because if disk usage goes back down then Cassandra will recover.

Rather, this is the first "transient" error that we're introducing.

The benefit of separating these is that it becomes clear to the operator that we don't need to worry about trying to fix an exceeded disk threshold error (whereas we do need to fix the corruptions).  However, I think we'll need to expose transient errors separately in the API so that we can pull these in the consuming products.